cypress-io/cypress
# fix: Clean up command attributes after no longer needed

### User facing changelog
No user-facing changes.

### Additional details
This PR adds a "cleanSubjects" step to the command_queue, which, every time a command resolves, does two things:
1. Looks through `cy.state('subjects')` and removes any subject chains that we know aren't needed by future commands. This should let us clean up the closures for executed queries, and drop unnecessary references to DOM elements that may no longer be attached to the page for commands.
2. Cleans up the attributes of passed commands, removing a bunch of data that we know we'll no longer display to the user or reference in the future.

### Steps to test
Existing tests should cover this fairly thoroughly; if it broke any features of the command log, they'll let us know.

If you wish to verify manually, the best place would be visually in the runner - examine the command log, verify that console props and snapshots continue to work for commands and tests that have already resolved.

### How has the user experience changed?
No change in user experience. Possibly less memory used in long tests with many commands executing.

### PR Tasks

- [n/a] Have tests been added/updated?
- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
